notify/notify.go

```diff
@@ -794,24 +809,69 @@ func (tms TimeMuteStage) Exec(ctx context.Context, l log.Logger, alerts ...*type
 		return ctx, alerts, errors.New("missing now timestamp")
 	}
 
-	muted := false
-Loop:
-	for _, mtName := range muteTimeIntervalNames {
-		mt, ok := tms.muteTimes[mtName]
-		if !ok {
-			return ctx, alerts, errors.Errorf("mute time %s doesn't exist in config", mtName)
-		}
-		for _, ti := range mt {
-			if ti.ContainsTime(now.UTC()) {
-				muted = true
-				break Loop
-			}
-		}
+	muted, err := inTimeIntervals(now, tms.Times, muteTimeIntervalNames)
+	if err != nil {
+		return ctx, alerts, err
 	}
+
 	// If the current time is inside a mute time, all alerts are removed from the pipeline.
 	if muted {
 		level.Debug(l).Log("msg", "Notifications not sent, route is within mute time")
 		return ctx, nil, nil
 	}
 	return ctx, alerts, nil
 }
+
+type TimeActiveStage timeStage
+
+func NewTimeActiveStage(ti map[string][]timeinterval.TimeInterval) *TimeActiveStage {
+	return &TimeActiveStage{ti}
+}
+
+// Exec implements the stage interface for TimeMuteStage.
```

Comment:
Minor typo, I think this should be TimeActiveStage?

Reply:
Thanks fixing the typo